src/test/scala/chiselTests/QueueSpec.scala

```diff
@@ -9,8 +9,8 @@ import chisel3.testers.BasicTester
 import chisel3.util._
```

Comment:
Overall I don't see what the new tests added are actually testing, other than the code still works? Shouldn't you also be checking for the existence of a SeqMem vs a Mem when the knob is set differently?

Reply:
You are right. The tests simply verify that both Mem and SyncReadMem implementations work. I agree with you that it would be good to check if that particular memory type exists, but I'm really not sure how it should be properly done. If you have any examples or hints, please share, and it'll be included.